ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -88,17 +88,15 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, NetworkMetr
    * @param pollTimeoutMs the poll timeout.
    * @return a list of {@link ResponseInfo} representing the responses received for any requests that were sent out
    * so far.
-   * @throws IOException if the {@link Selector} associated with this NetworkClient throws
```

Comment:
But what are we trying to isolate here? 

Also, if the exception is indeed fatal, this will immediately create a busy waiting loop.